### PR TITLE
Address eellison's CR comments on AOTAutograd

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85370

For some reason, this change didn't actually make it to master.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

### CODE REVIEW

File: functorch/_src/aot_autograd.py [hunk 1]
```diff
@@ -312,25 +312,31 @@ def aot_dispatch_autograd(flat_fn, flat_args: List[Tensor], aot_config: AOTConfi
...
+    # NB: Hot path, avoid set lookups here
     def remove_dupe_args(args):
+        if not dropped_args:
+            return args
         r = []
-        for i, t in enumerate(args):
-            if i in drop_args:
-                continue
-            r.append(t)
+        for t, keep in zip(args, keep_arg_mask):
```
Comment (vadimkantorov) [inline functorch/_src/aot_autograd.py:337]: is simpler as a list comprehension? `return [t for t, keep in zip(args, keep_arg_mask) if keep]`

same for a function below?

Comment (ezyang) [reply functorch/_src/aot_autograd.py:337]: sure

**Review by eellison — APPROVED** [review]

**Pull request closed without merging**